Src/coordinate.F90

```diff
@@ -2307,42 +2308,57 @@ subroutine SetNetwork(ipt)
    ipclbeg = 0
    ipclend = 0
 
+   if(.not.allocated(rorigin)) then
+      allocate(rorigin(3,nnw))
+      rorigin = 0.0E+00
+   end if
+
 ! ... set gels
 
    maxnbondcl(ipt) = nclnode
+   inwglob = sum(nnwnwt(1:inwt-1))
 
    do inw = 1, nnwnwt(inwt)! loop over all networks of network type inwt
+      inwglob = inwglob + 1
+
 try:  do itry = 1, ntry    ! loop over attempts to set the gel
 
 !  ... set network origin
 
          if (txoriginnwt(inwt) == 'origin') then
-            rorigin = Zero
+            rorigin(1:3,inwglob) = Zero
          else if (txoriginnwt(inwt) == 'random') then
             if (lbcsph) then
                do
-                  rorigin(1) = Two*sphrad*(Random(iseed) - Half)
-                  rorigin(2) = Two*sphrad*(Random(iseed) - Half)
-                  rorigin(3) = Two*sphrad*(Random(iseed) - Half)
-                  if (sum(rorigin(1:3)**2) <= sphrad2) exit
+                  rorigin(1,inwglob) = Two*sphrad*(Random(iseed) - Half)
+                  rorigin(2,inwglob) = Two*sphrad*(Random(iseed) - Half)
+                  rorigin(3,inwglob) = Two*sphrad*(Random(iseed) - Half)
+                  if (sum(rorigin(1:3,inwglob)**2) <= sphrad2) exit
                end do
             else if(lbcbox) then
-               rorigin(1) = boxlen(1)*(Random(iseed) - Half)
-               rorigin(2) = boxlen(2)*(Random(iseed) - Half)
-               rorigin(3) = boxlen(3)*(Random(iseed) - Half)
-               call PBC(rorigin(1), rorigin(2), rorigin(3))
+               rorigin(1,inwglob) = boxlen(1)*(Random(iseed) - Half)
+               rorigin(2,inwglob) = boxlen(2)*(Random(iseed) - Half)
+               rorigin(3,inwglob) = boxlen(3)*(Random(iseed) - Half)
+               call PBC(rorigin(1,inwglob), rorigin(2,inwglob), rorigin(3,inwglob))
             else
                call Stop (txroutine, 'txbc is not compatible with origin position of network', uout)
             end if
          else
             call Stop(txroutine,'invalid choice of txoriginnwt',uout)
          end if
 
+! ... test whether interpenetration with already set networks occurs
+
+         do jnwglob = 1, nnw
```

Comment:
Do we have to loop up to `nnw`? Shouldn't it be enough to loop up to `inwglob`?

If it tries to set a network close to 0,0,0 this will fail when no other networks have been set, as for all unset networks, `rorigin = 0.0`

Reply:
Yes, we have to loop up to `nnw`. Each time subroutine `SetNetwork` is being called, all networks of one network type are being set. However, the order in which different network types are set here, is not neccessarily consecutive, as the currently considered network type depends on the particle type `ipt`, which subroutine `SetNetwork` receives.